src/anyon/anyon.jl

```diff
@@ -1,5 +1,7 @@
 using Snowflurry
 
+abstract type AbstractAnyonQPU <: AbstractQPU end
```

Comment:
This is using polymorphism to share code. I would prefer if we use composition.

The polymorphism isn't inherently wrong, as both MonarQ and Yukon are AnyonQPUs. However, AnyonQPUs will continue to evolve where MonarQ and Yukon won't, as they have different evolution paths. This will cause friction to change AnyonQPU while not breaking MonardQ and Yukon. We will have a remote virtual QPU soon, which is also an AnyonQPU but won't be limited to Yukon and MonarQ's gate set.

What is more long-term is the usage of AnyonClient, but that will also only be true until we, at some point, release version 2 of our API.

Sharing code using inheritance couples the evolution of the subclasses to the base class. This might be the intent. In the classic, `Dog` and `Cat` are both `Animal`s example, I would be extremely surprised if  `Animal` evolves in such a way that makes it incompatible with `Cat` or `Dog`. Using inheritance in that example makes a lot of sense.

For example, we can define a function, `run_circuit_using_client`, that implements `run_job`, but with an `AnyonClient`, since after transpilation and validation only the client is used to send a job and wait for the result. This will allow both `AnyonYukonQPU` and `AnyonMonarQQPU` to use this method, but without coupling them to anything.

We can also factor out the CZ gateset functions and use that from both `AnyonYukonQPU` and `AnyonMonarQQPU`.

If anything is unclear, we can do a teams call tomorrow.